src/store/slices/userSlice.ts

```diff
@@ -20,9 +20,8 @@ const userSlice = createSlice({
   name: "user",
   initialState,
   reducers: {
-    setAuthToken: (state, action) => {
-      const { authToken, email, id } = action.payload;
-      return { ...state, authToken, email, id };
+    setUser: (state, action) => {
+      return { ...state, ...action.payload };
     },
     setSignIn: (state, action) => {
```

Comment:
we need types for state, action as I 'm not sure wat data are you storing

Reply:
I've simplified the reducers, now only setUser takes care of modifying User related states like id, role, authToken, etc. 

The idea now, as you can see below, is that we can set any User related states by passing an object with only the desired fields in the dispatcher. If in the future we add, for example, first and last names as persisted states this will be easy to modify and use.

I've also added types to the payload. The state types are already inferred by the initial state object
```typescript
interface UserState {
  authToken?: string | undefined;
  email?: string | undefined;
  role?: "freelancer" | "client" | undefined;
  id?: number | undefined;
}
{
  setUser: (state, { payload }: PayloadAction<UserState>) => {
    return { ...state, ...payload };
  },
}
```

Reply:
hmm its fine, but do we need to store id? where will we use it?

Reply:
I'm using the id in the mutation below to update the roles field in the database. I could, instead of a PATCH request, send a POST with only the email and role in the body to the backend. What do you think? Is storing the id a bad practice?

```typescript
  const sendToDB = () => {
    updateRole({
      id,
      role,
    });
    navigate(SETTINGS_ROUTE);
  };


//RTK query setup:
const userApi = createApi({
  reducerPath: "setUser",
  baseQuery: fetchBaseQuery({ baseUrl: process.env.REACT_APP_API_URL }),
  endpoints: (builder) => ({
    setUser: builder.mutation<User, User>({
      query: ({ id, ...patch }) => ({
        url: `/settings-info/${id}`,
        method: "PATCH",
        body: {
          ...patch,
        },
      }),
      transformResponse: (response: User) => response,
    }),
  }),
});
```

Reply:
but why you dont use bearer token its not secure to use id

Reply:
Yes, it's missing the Bearer token, I'll implement it.

Reply:
@lyudad I've used the jwtDecode package to extract the user id from the token since it is being sent in the payload from the backend. Also removed the id from the user global state and updated the Contact Info form as well, since it had the same problem of using the id persisted state in its requests.